internal/pkg/cli/deploy/workload.go

```diff
@@ -370,24 +376,29 @@ func (d *workloadDeployer) uploadContainerImages(imgBuilderPusher imageBuilderPu
 
 func buildArgsPerContainer(name, workspacePath string, img ContainerImageIdentifier, unmarshaledManifest interface{}) (map[string]*dockerengine.BuildArguments, error) {
 	type dfArgs interface {
-		BuildArgs(rootDirectory string) map[string]*manifest.DockerBuildArgs
+		BuildArgs(rootDirectory string) (map[string]*manifest.DockerBuildArgs, error)
 		ContainerPlatform() string
 	}
 	mf, ok := unmarshaledManifest.(dfArgs)
 	if !ok {
 		return nil, fmt.Errorf("%s does not have required methods BuildArgs() and ContainerPlatform()", name)
 	}
-	var tags []string
-	tags = append(tags, imageTagLatest, img.ReferenceTag())
-	args := mf.BuildArgs(workspacePath)
-	dArgs := make(map[string]*dockerengine.BuildArguments, len(args))
-	for k, v := range args {
-		dArgs[k] = &dockerengine.BuildArguments{
-			Dockerfile: aws.StringValue(v.Dockerfile),
-			Context:    aws.StringValue(v.Context),
-			Args:       v.Args,
-			CacheFrom:  v.CacheFrom,
-			Target:     aws.StringValue(v.Target),
+	argsPerContainer, err := mf.BuildArgs(workspacePath)
+	if err != nil {
+		return nil, fmt.Errorf("check if manifest requires building from local Dockerfile: %w", err)
+	}
+	dArgs := make(map[string]*dockerengine.BuildArguments, len(argsPerContainer))
+	tags := []string{imageTagLatest, img.Tag()}
+	for container, buildArgs := range argsPerContainer {
+		if container != name {
+			tags = []string{fmt.Sprintf("%s-%s", container, imageTagLatest), fmt.Sprintf("%s-%s", container, img.nonCustomTag())}
```

Comment:
Thinking about this, every new run of `copilot svc package --upload-assets` will result in a new UUID in the constructor.

Wouldn't that mean each time we run the command if the user has a `<sidecar>.image.build` then we will end up pushing a new tag for the same sidecar image even if nothing changed? That seems problematic 💭

Reply:
This also happens when they are `no infrastructure changes` when user runs `copilot svc deploy` everytime a new random UUID tag will be attached. What could be potential solution for this 🤔